src/main/java/org/gradle/crypto/checksum/Checksum.java

```diff
@@ -88,20 +86,40 @@ public void setFiles(FileCollection files) {
         this.files = files;
     }
 
-    @Input
+    /**
+     * @return algorithm
+     * @deprecated Use getAlgorithmProperty() instead
+     */
+    @Deprecated
+    @Internal
     public Algorithm getAlgorithm() {
         return algorithm.get();
     }
 
+    @Input
+    public Property<Algorithm> getAlgorithmProperty() {
+        return algorithm;
+    }
+
     public void setAlgorithm(Algorithm algorithm) {
         this.algorithm.set(algorithm);
     }
 
-    @OutputDirectory
+    /**
+     * @return output directory
+     * @deprecated Use getOutputDirProperty() instead
+     */
+    @Deprecated
+    @Internal
     public File getOutputDir() {
         return outputDir.get().getAsFile();
     }
 
+    @OutputDirectory
+    public DirectoryProperty getOutputDirProperty() {
```

Comment:
Nit pick: I would name the new getter `getOutputDirectory()` and `getChecksumAlgorithm()` instead

Reply:
Sure, better than leaking internal details.
out of curiosity, why are those getters needed?

Reply:
These getters are required so the task properties can be configured.